### PR TITLE
#23 - fix position + fix removing parent publisher container

### PR DESCRIPTION
**Pull request by mark-veenstra**
`getPosition()` added margins from the container and itself, therefore the video was not displayed correctly. 

`unpublish()` also removed the parent container from where it was appended, therefore this DOM element is missing in the view when inserting a publisher back again.

### CODE REVIEW

File: src/js/OTSession.coffee [hunk 1]
```diff
@@ -110,7 +110,10 @@ class TBSession
     console.log("JS: Unpublish")
     element = document.getElementById( @publisher.domId )
     if(element)
-      element.parentNode.removeChild(element)
+      element.firstChild && element.removeChild(element.firstChild);
```
Comment (msach22) [inline src/js/OTSession.coffee]: you can use `this.resetElement` to accomplish what you're adding here

File: src/js/OTSession.coffee [hunk 2]
```diff
@@ -136,8 +139,11 @@ class TBSession
     objects = document.getElementsByClassName('OT_root')
     while( objects.length > 0 )
       e = objects[0]
-      if e and e.parentNode and e.parentNode.removeChild
-        e.parentNode.removeChild(e)
+      if e and e.removeChild
```
Comment (msach22) [inline src/js/OTSession.coffee]: I believe `this.resetElement` is doing the same

File: src/js/OTSession.coffee (hunk 1 above)
Comment (mark-veenstra) [reply src/js/OTSession.coffee]: Adjusted and used the helper method

File: src/js/OTSession.coffee (hunk 2 above)
Comment (mark-veenstra) [reply src/js/OTSession.coffee]: Adjusted and used the helper method

Comment (msach22) [reply src/js/OTSession.coffee]: Sorry to be a pain @mark-veenstra but wouldn't `e.removeChild` require a node to be passed in. IMHO, I think you just have to see if `e` exists b/c `resetElement` will take care of the children

Comment (mark-veenstra) [reply src/js/OTSession.coffee]: @msach22 you are right. Changed the commit.

**Pull request merged**